app/models/concerns/fog_extensions/debug/debug_http_instrumentor.rb

```diff
@@ -0,0 +1,67 @@
+module FogExtensions
+  module Debug
+    class DebugHttpInstrumentor
+      class << self
+        attr_accessor :events
+
+        def instrument(name, params = {}, &block)
```

Comment:
Rather than writing something against the instrumentation API itself, it'd might be preferable to pass in `ActiveSupport::Notifications` and then write against `ActiveSupport::LogSubscriber`.

See config/initializers/ldap_instrumentation.rb for an example.  Just add another initialiser that attaches to the names of the events we're interested in?